octoprint/datadog_checks/octoprint/check.py

```diff
@@ -0,0 +1,138 @@
+            )
+            self.log.debug('rpi core temp - temp: %s', temp)
+            self.log.debug('rpi core temp - err: %s', err)
+            self.log.debug('rpi core temp - retcode: %s', retcode)
+            temp = temp.replace("temp=", "").replace("'C", "")
+            if temp.startswith("VCHI initialization failed"):
+                self.log.info(
+                    "Unable to get rPi temp.  To resolve, add the 'video' group to the 'dd-agent' user"
+                    " by running `sudo usermod -aG video dd-agent`"
+                )
+                temp = 0.0
+        else:
+            self.log.info(
+                "The command typically used to get the core temperature, /usr/bin/vcgencmd,"
+                "is not available on this system."
+            )
+            temp = 0.0
+        return float(temp)
+
+    def seconds_to_minutes(self, seconds):
+        if not seconds:
+            return 0
+        else:
+            return int(seconds / 60)
+
+    # Get stats from REST API as json
+    def get_job_info(self, server, key, timeout):
+        PATH = "/api/job"
+        url = server + PATH
+        headers = {"X-Api-Key": key, "content-type": "application/json"}
+        req = requests.get(url, timeout=timeout, headers=headers)
+        return req.json()
+
+    def get_tool_temp(self, server, key, timeout):
+        PATH = "/api/printer/tool"
+        url = server + PATH
+        headers = {"X-Api-Key": key, "content-type": "application/json"}
+        req = requests.get(url, timeout=timeout, headers=headers)
+        return req.json()
+
+    def get_bed_temp(self, server, key, timeout):
+        PATH = "/api/printer/bed"
+        url = server + PATH
+        headers = {"X-Api-Key": key, "content-type": "application/json"}
+        req = requests.get(url, timeout=timeout, headers=headers)
+        return req.json()
+
+    def check(self, instance):
+        octo_api_key = instance.get('octo_api_key')
+        self.log.debug('octo_api_key (from config): %s', octo_api_key)
+        self.log.debug('octo_server: %s', SERVER)
+        rpi_core_temp = self.get_rpi_core_temp()
+        self.gauge("octoprint.rpi_core_temp", rpi_core_temp)
+
+        # get api data
+        job_info = self.get_job_info(SERVER, octo_api_key, TIMEOUT)
+
+        # # Job State
+        state = job_info["state"]
+        # States: Printing, Paused, Cancelled, Operational...
+        # The proper thing would be a a case/switch statement
+        if state == "Operational":
+            printer_state = 0
+        elif state == "Paused":
+            printer_state = 1
+        elif state == "Printing":
+            printer_state = 2
+        else:
+            printer_state = -1
+        self.gauge("octoprint.printer_state", printer_state)
+
+        # Print Job Percent Completed and Time Estimate
+        est_print_time = self.seconds_to_minutes(job_info["job"]["estimatedPrintTime"])
```

Comment:
Seconds would be a higher resolution metric to report instead of minutes, and is also the unit in `metadata.csv`.

Reply:
Agree that seconds is more granular, but most print jobs take hours, if not days.  Great catch that I have it wrong in `metadata.csv`!  Thanks!